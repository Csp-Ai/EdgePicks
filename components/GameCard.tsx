import React, { useRef } from 'react';
import Image from 'next/image';
import type { Game } from '@/lib/types';
import { formatKickoff } from '@/lib/utils/formatKickoff';

interface Props {
  game: Game;
  onClick: (game: Game) => void;
  onHover?: () => void;
}

const GameCard: React.FC<Props> = ({ game, onClick, onHover }) => {
  const kickoffRelative = game.kickoffDisplay ?? formatKickoff(game.time);
  const kickoffAbsolute = new Date(game.time).toLocaleTimeString([], {
    hour: 'numeric',
    minute: '2-digit',
  });
  const hoverRef = useRef<ReturnType<typeof setTimeout> | null>(null);

  const handleEnter = () => {
    hoverRef.current = setTimeout(() => onHover?.(), 200);
  };

  const handleLeave = () => {
    if (hoverRef.current) {
      clearTimeout(hoverRef.current);
      hoverRef.current = null;
    }
  };

  return (
    <button
      onClick={() => onClick(game)}
      onMouseEnter={handleEnter}
      onFocus={handleEnter}
      onMouseLeave={handleLeave}
      onBlur={handleLeave}
      className={`card group w-full text-left bg-slate-800 hover:bg-slate-700 focus:outline-none focus-visible:ring-2 focus-visible:ring-emerald-500 ring-1 ring-slate-700 transition-colors`}
      aria-label={`Analyze ${game.homeTeam} vs ${game.awayTeam} kickoff ${kickoffAbsolute}`}
    >
      <div className="flex items-center justify-between mb-3 text-slate-100">
        <div className="flex items-center gap-2">
          {game.homeLogo && (
            <Image
              src={game.homeLogo}
              alt={`${game.homeTeam} logo`}
              width={24}
              height={24}
            />
          )}
          <span>{game.homeTeam}</span>
        </div>
        <span className="text-slate-300">vs</span>
        <div className="flex items-center gap-2">
          {game.awayLogo && (
            <Image
              src={game.awayLogo}
              alt={`${game.awayTeam} logo`}
              width={24}
              height={24}
            />
          )}
          <span>{game.awayTeam}</span>
        </div>
      </div>
      <div className="text-xs text-slate-300">{kickoffRelative} â€¢ {kickoffAbsolute}</div>
      {game.odds ? (
        <div className="flex flex-wrap gap-1 mt-2 text-[11px]">
          {game.odds.spread !== undefined && (
            <span className="px-2 py-0.5 rounded-full border border-emerald-700/40 bg-emerald-900/20 text-emerald-100" title="Spread">
              Spread {game.odds.spread}
            </span>
          )}
          {game.odds.overUnder !== undefined && (
            <span className="px-2 py-0.5 rounded-full border border-emerald-700/40 bg-emerald-900/20 text-emerald-100" title="Over/Under">
              O/U {game.odds.overUnder}
            </span>
          )}
          {game.odds.moneyline && (
            <span className="px-2 py-0.5 rounded-full border border-emerald-700/40 bg-emerald-900/20 text-emerald-100" title="Moneyline">
              ML {game.odds.moneyline.home ?? ''}
              {game.odds.moneyline.home && game.odds.moneyline.away ? ' / ' : ''}
              {game.odds.moneyline.away ?? ''}
            </span>
          )}
        </div>
      ) : (
        <div className="text-xs text-slate-400 mt-2">No odds yet</div>
      )}
      {game.odds?.bookmaker && (
        <div className="mt-1 text-[10px] text-slate-400">via {game.odds.bookmaker}</div>
      )}
    </button>
  );
};

export default GameCard;

