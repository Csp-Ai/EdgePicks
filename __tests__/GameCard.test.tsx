import React from 'react';
import { render, screen } from '@testing-library/react';
import GameCard from '../components/GameCard';
import type { Game } from '../lib/types';

jest.mock('next/image', () => (props: any) => {
  // eslint-disable-next-line @next/next/no-img-element
  return <img {...props} />;
});

const game: Game = {
  id: 'game-1', // Added missing id property
  gameId: '1',
  league: 'NFL',
  homeTeam: 'Lakers',
  awayTeam: 'Celtics',
  time: '2023-01-01T19:20:00Z',
  homeLogo: '/home.png',
  awayLogo: '/away.png',
  odds: {
    homeSpread: -3.5, // Replaced `spread` with `homeSpread`
    awaySpread: 3.5, // Added `awaySpread`
    total: 45.5, // Replaced `overUnder` with `total`
  },
};

describe('GameCard', () => {
  it('renders kickoff info and odds chips', () => {
    render(<GameCard game={game} onClick={jest.fn()} />);
    const absolute = new Date(game.time).toLocaleTimeString([], { hour: 'numeric', minute: '2-digit' });
    const kickoff = screen.getByText(/in 2h/);
    expect(kickoff).toBeInTheDocument();
    expect(kickoff.textContent).toContain(`â€¢ ${absolute}`);
    expect(screen.getByText(/Spread/)).toBeInTheDocument();
    expect(screen.getByText(/O\/U/)).toBeInTheDocument();
    expect(screen.getByText(/ML/)).toBeInTheDocument();
  });

  it('has accessible button role', () => {
    render(<GameCard game={game} onClick={jest.fn()} />);
    expect(screen.getByRole('button')).toBeInTheDocument();
  });

  it('matches contrast snapshot', () => {
    const { container } = render(<GameCard game={game} onClick={jest.fn()} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
