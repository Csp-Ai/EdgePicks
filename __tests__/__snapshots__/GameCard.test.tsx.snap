// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameCard matches contrast snapshot 1`] = `
<button
  aria-label="Analyze Lakers vs Celtics kickoff 7:20 PM"
  class="card group w-full text-left bg-slate-800 hover:bg-slate-700 focus:outline-none focus-visible:ring-2 focus-visible:ring-emerald-500 ring-1 ring-slate-700 transition-colors"
>
  <div
    class="flex items-center justify-between mb-3 text-slate-100"
  >
    <div
      class="flex items-center gap-2"
    >
      <img
        alt="Lakers logo"
        height="24"
        src="/home.png"
        width="24"
      />
      <span>
        Lakers
      </span>
    </div>
    <span
      class="text-slate-300"
    >
      vs
    </span>
    <div
      class="flex items-center gap-2"
    >
      <img
        alt="Celtics logo"
        height="24"
        src="/away.png"
        width="24"
      />
      <span>
        Celtics
      </span>
    </div>
  </div>
  <div
    class="text-xs text-slate-300"
  >
    in 2h
     â€¢ 
    7:20 PM
  </div>
  <div
    class="flex flex-wrap gap-1 mt-2 text-[11px]"
  >
    <span
      class="px-2 py-0.5 rounded-full border border-emerald-700/40 bg-emerald-900/20 text-emerald-100"
      title="Spread"
    >
      Spread 
      -3.5
    </span>
    <span
      class="px-2 py-0.5 rounded-full border border-emerald-700/40 bg-emerald-900/20 text-emerald-100"
      title="Over/Under"
    >
      O/U 
      45.5
    </span>
    <span
      class="px-2 py-0.5 rounded-full border border-emerald-700/40 bg-emerald-900/20 text-emerald-100"
      title="Moneyline"
    >
      ML 
      -150
       / 
      130
    </span>
  </div>
  <div
    class="mt-1 text-[10px] text-slate-400"
  >
    via 
    MockBook
  </div>
</button>
`;
